crypto/src/proofs/transparent.rs

```diff
@@ -466,83 +465,63 @@ impl SwapProof {
             return Err(Error::BadNullifier);
         }
 
-        // Spend authority.
-        let rk_bytes: [u8; 32] = rk.into();
-        let rk_test = self.ak.randomize(&self.spend_auth_randomizer);
-        let rk_test_bytes: [u8; 32] = rk_test.into();
-        if rk_bytes != rk_test_bytes {
-            return Err(Error::InvalidSpendAuthRandomizer);
-        }
-
-        // Diversified address integrity.
-        let fvk = keys::FullViewingKey::from_components(self.ak, self.nk);
-        let ivk = fvk.incoming();
-        if self.pk_d != ivk.diversified_public(&self.g_d) {
-            return Err(Error::InvalidDiversifiedAddress);
-        }
+        // TODO: is there an address check that needs to occur?
```

Comment:
Yes, not just an address check but also the checks that are part of an `Output` proof.  We need to:

- create two output notes and do all the normal output checks;
- prove that one output is $\Lambda_1$ of asset $t_1$;
- prove that the other is $\Lambda_2$ of asset $t_2$;
- prove that both notes are sent to the transmission key committed to in the swap NFT's asset ID.

Reply:
We could leave out the amount correctness statements for now, because we don't currently have a way to model the price data that should be public inputs to the proof (that's #1126), and it would be good to merge a preliminary version of this first